examples/Migration/CampaignManagement/CreateCompleteCampaignAdwordsApiOnly.php

```diff
@@ -17,12 +17,364 @@
+use Google\AdsApi\AdWords\v201809\cm\AdGroupCriterionService;
+use Google\AdsApi\AdWords\v201809\cm\AdGroupOperation;
+use Google\AdsApi\AdWords\v201809\cm\AdGroupService;
+use Google\AdsApi\AdWords\v201809\cm\AdGroupStatus;
+use Google\AdsApi\AdWords\v201809\cm\AdRotationMode;
+use Google\AdsApi\AdWords\v201809\cm\AdvertisingChannelType;
+use Google\AdsApi\AdWords\v201809\cm\BiddableAdGroupCriterion;
+use Google\AdsApi\AdWords\v201809\cm\BiddingStrategyConfiguration;
+use Google\AdsApi\AdWords\v201809\cm\BiddingStrategyType;
+use Google\AdsApi\AdWords\v201809\cm\Budget;
+use Google\AdsApi\AdWords\v201809\cm\BudgetBudgetDeliveryMethod;
+use Google\AdsApi\AdWords\v201809\cm\BudgetOperation;
+use Google\AdsApi\AdWords\v201809\cm\BudgetService;
+use Google\AdsApi\AdWords\v201809\cm\Campaign;
+use Google\AdsApi\AdWords\v201809\cm\CampaignOperation;
+use Google\AdsApi\AdWords\v201809\cm\CampaignService;
+use Google\AdsApi\AdWords\v201809\cm\CampaignStatus;
+use Google\AdsApi\AdWords\v201809\cm\CpcBid;
+use Google\AdsApi\AdWords\v201809\cm\ExpandedTextAd;
+use Google\AdsApi\AdWords\v201809\cm\Keyword;
+use Google\AdsApi\AdWords\v201809\cm\KeywordMatchType;
+use Google\AdsApi\AdWords\v201809\cm\Money;
+use Google\AdsApi\AdWords\v201809\cm\NetworkSetting;
+use Google\AdsApi\AdWords\v201809\cm\Operator;
+use Google\AdsApi\AdWords\v201809\cm\UrlList;
+use Google\AdsApi\AdWords\v201809\cm\UserStatus;
+
 require __DIR__ . '/../vendor/autoload.php';
 
+/**
+ * This code example is the first in a series of code examples that shows how to create
+ * a Search campaign using the AdWords API, and then migrate it to Google Ads API one
+ * functionality at a time. See other examples in this directory for code examples in various
+ * stages of migration.
+ *
+ * This code example represents the initial state, where the AdWords API is used to create a
+ * campaign budget, a Search campaign, ad groups, keywords and expanded text ads. None of the
+ * functionality has yet been migrated to the Google Ads API.
+ */
 class CreateCompleteCampaignAdwordsApiOnly
 {
-    public static function runExample()
-    {
-        echo "Not implemented yet";
+    // Number of ads being added/updated in this code example.
+    private const NUMBER_OF_ADS = 5;
+    // The list of keywords being added in this code example.
+    private const KEYWORDS_TO_ADD = [
+        "mars cruise",
+        "space hotel"
+    ];
+
+    /**
+     * Runs the CreateCompleteCampaignGoogleAdsApiOnly example.
+     * @param AdWordsServices $adWordsServices the AdWords services
+     * @param AdWordsSession $adWordsSession the AdWords session
+     */
+    public static function runExample(
+        AdWordsServices $adWordsServices,
+        AdWordsSession $adWordsSession
+    ) {
+        $campaignBudget = self::createCampaignBudget($adWordsServices, $adWordsSession);
+        $campaign =
+            self::createCampaign($adWordsServices, $adWordsSession, $campaignBudget->getBudgetId());
```

Comment:
Why are you passing only the budget id and then, inside `createCampaign`, creating a `new Budget()` setting its id to `$budgetId`?

Wouldn't it be more concise and, in my opinion, clearer, to pass the whole campaignBudget object as a parameter to the createCampaign function and set it as the campaign budget for the campaign being created?

(The same applies to the campaign parameter in `createAdGroup` and the adGroup parameter in `createTextAds` and `createKeywords`)

Reply:
Nevermind, I just realized it makes it easier for the "hybrid" examples to pass the object ids as parameter because you'd end up having a Google Ads API AdGroup object in later stages rather than an AdWords API AdGroup object, so passing the id is better.